static/js/components/email/hoc.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import R from 'ramda';
+
+import EmailCompositionDialog from './EmailCompositionDialog';
+import { getAllEmailProperties } from './lib';
+import { getDisplayName } from '../../util/util';
+
+export const withEmailDialog = R.curry((emailType, WrappedComponent) => {
+  class WithEmailDialog extends React.Component {
+    constructor(props) {
+      super(props);
+      this.renderCompositionDialog = this.renderCompositionDialog.bind(this);
+      this.allEmailProperties = getAllEmailProperties();
+      this.emailType = emailType;
+    }
+
+    renderCompositionDialog() {
+      let {
+        dispatch,
+        ui: { dialogVisibility },
+        email
+      } = this.props;
+      return <EmailCompositionDialog
+        dispatch={dispatch}
+        dialogVisibility={dialogVisibility}
+        email={email}
+        emailProperties={this.allEmailProperties}
+      />;
+    }
+
+    render() {
+      let emailDialog = this.props.emailDialog || this.renderCompositionDialog();
+
+      let emailOpenActions = this.props.emailOpenActions || {};
+      emailOpenActions[this.emailType] = R.partial(
+        this.allEmailProperties[this.emailType].emailOpenAction, [this.props.dispatch]
+      );
+
+      return <WrappedComponent
+        {...this.props}
+        emailDialog={emailDialog}
+        emailOpenActions={emailOpenActions}
+      />;
+    }
+  }
+
+  if (WithEmailDialog.displayName && !WithEmailDialog.displayName.startsWith('WithEmailDialogs')) {
```

Comment:
my original plan was to include multiple email dialogs on a page by invoking `withEmailDialog()` multiple times, but i'm realizing now that doesn't make as much sense as just invoking it once with an array of email types (if you need more than one). this is going to change. aside: if you're up for the annoyance, i'd love to have you as a second reviewer!